test/test_ekfac.py

```diff
@@ -0,0 +1,1094 @@
+    y = rand(batch_size, D_out)
+    model = Linear(D_in, D_out)
+
+    params = list(model.parameters())
+    # create and compute EKFAC
+    ekfac = EKFACLinearOperator(
+        model,
+        MSELoss(reduction="sum"),
+        params,
+        [(X, y)],
+    )
+
+    # save state dict
+    state_dict = ekfac.state_dict()
+    EKFAC_PATH = "ekfac_state_dict.pt"
+    save(state_dict, EKFAC_PATH)
+
+    # create new EKFAC with different loss function and try to load state dict
+    ekfac_new = EKFACLinearOperator(
+        model,
+        CrossEntropyLoss(),
+        params,
+        [(X, y)],
+    )
+    with raises(ValueError, match="loss"):
+        ekfac_new.load_state_dict(load(EKFAC_PATH, weights_only=False))
+
+    # create new EKFAC with different loss reduction and try to load state dict
+    ekfac_new = EKFACLinearOperator(
+        model,
+        MSELoss(),
+        params,
+        [(X, y)],
+    )
+    with raises(ValueError, match="reduction"):
+        ekfac_new.load_state_dict(load(EKFAC_PATH, weights_only=False))
+
+    # create new EKFAC with different model and try to load state dict
+    wrong_model = Sequential(Linear(D_in, 10), ReLU(), Linear(10, D_out))
+    wrong_params = list(wrong_model.parameters())
+    ekfac_new = EKFACLinearOperator(
+        wrong_model,
+        MSELoss(reduction="sum"),
+        wrong_params,
+        [(X, y)],
+    )
+    with raises(RuntimeError, match="loading state_dict"):
+        ekfac_new.load_state_dict(load(EKFAC_PATH, weights_only=False))
+
+    # create new EKFAC and load state dict
+    ekfac_new = EKFACLinearOperator(
+        model,
+        MSELoss(reduction="sum"),
+        params,
+        [(X, y)],
+        check_deterministic=False,  # turn off to avoid computing EKFAC again
+    )
+    ekfac_new.load_state_dict(load(EKFAC_PATH, weights_only=False))
+    # clean up
+    os.remove(EKFAC_PATH)
+
+    # check that the two EKFACs are equal
+    compare_state_dicts(ekfac.state_dict(), ekfac_new.state_dict())
+    test_vec = rand(ekfac.shape[1])
+    report_nonclose(ekfac @ test_vec, ekfac_new @ test_vec)
+
+
+def test_from_state_dict():
+    """Test that EKFACLinearOperator can be created from state dict."""
+    manual_seed(0)
+    batch_size, D_in, D_out = 4, 3, 2
+    X = rand(batch_size, D_in)
+    y = rand(batch_size, D_out)
+    model = Linear(D_in, D_out)
+
+    params = list(model.parameters())
+    # create and compute EKFAC
+    ekfac = EKFACLinearOperator(
+        model,
+        MSELoss(reduction="sum"),
+        params,
+        [(X, y)],
+    )
+
+    # save state dict
+    state_dict = ekfac.state_dict()
+
+    # create new EKFAC from state dict
+    kfac_new = EKFACLinearOperator.from_state_dict(state_dict, model, params, [(X, y)])
+
+    # check that the two EKFACs are equal
+    compare_state_dicts(ekfac.state_dict(), kfac_new.state_dict())
+    test_vec = rand(ekfac.shape[1])
+    report_nonclose(ekfac @ test_vec, kfac_new @ test_vec)
```

Comment:
Could these tests also share code with their KFAC equivalents?

Reply:
Could be worth to refactor them into a more general function that tests whether a stateful linear operator can be successfully saved and loaded from its state dict.